pool.go

```diff
@@ -0,0 +1,162 @@
+package websocket
+
+import (
+	"bufio"
+	"errors"
+	"sync"
+)
+
+// Pool is a general-purpose interface for pooling.
+// Pool is implemented by sync.Pool.
+type Pool interface {
```

Comment:
'Cause this is an interface for a specific reason(byte pool), maybe we don't need to have params/results as an `interface{}` WDYT?

Reply:
2 reasons:
1. It actually isnt - it is also used to pool *bufio.Reader
2. It is implemented by sync.Pool